interface/modules/custom_modules/oe-module-comlink-telehealth/src/Controller/TeleHealthFrontendSettingsController.php

```diff
@@ -0,0 +1,70 @@
+        $this->assetPath = $assetPath;
+        $this->twig = $twig;
+    }
+
+    public function renderFrontendSettings()
+    {
+        $assetPath = $this->assetPath;
+        // strip off the assets, and public folder to get to the base of our module directory
+        $modulePath = dirname(dirname($assetPath)) . "/"; // make sure to end with a path
+        echo $this->twig->render("comlink/telehealth-frontend-settings.js.twig", [
+            'settings' => [
+                'translations' => $this->getTranslationSettings()
+                ,'modulePath' => $modulePath
+                ,'assetPath' => $assetPath
+            ]
+        ]);
+    }
+    public function getTranslationSettings()
+    {
+        $translations = [
+                'CALL_CONNECT_FAILED' => xlt("Failed to connect the call."),
+                'SESSION_LAUNCH_FAILED' => xlt("There was an error in launching your telehealth session.  Please try again or contact support"),
+                'DUPLICATE_SESSION' => xlt("You are already in a conference session.  Please hangup the current call to start a new telehealth session"),
+                'HOST_LEFT' => xlt("Host left the call"),
+                'PROVIDER_SESSION_START_PROMPT' => xlt("Would you like to start a telehealth session with this patient (This will create an encounter if one does not exist)?"),
+                'PROVIDER_SESSION_CLONE_START_PROMPT' => xlt("This appointment belongs to a different provider. Would you still like to start a telehealth session (this will copy the appointment to your calendar and create an encounter if needed)?"),
+                'PROVIDER_SESSION_TELEHEALTH_UNENROLLED' => xlt("This is a Telehealth session appointment.  If you would like to provide telehealth sessions to your clients contact your administrator to enroll today."),
+                'CONFIRM_SESSION_CLOSE' => xlt("Are you sure you want to close this session?"),
+                "TELEHEALTH_MODAL_TITLE" => xlt("TeleHealth Session"),
+                "TELEHEALTH_MODAL_CONFIRM_TITLE" => xlt("Confirm Session Close"),
+                "UPDATE_APPOINTMENT_STATUS" => xlt("Update appointment status"),
+                "STATUS_NO_SHOW" => xlt("No Show"),
+                "STATUS_CANCELED" => xlt("Canceled"),
+                "STATUS_CHECKED_OUT" => xlt("Checked Out"),
+                "CONFIRM" => xlt("Confirm"),
+                "CALENDAR_EVENT_DISABLED" => xlt("TeleHealth Sessions can only be launched within two hours of the current appointment time."),
+                "CALENDAR_EVENT_COMPLETE" => xlt("This TeleHealth appointment has been completed."),
+                "STATUS_SKIP_UPDATE" => xlt("Skip Update"),
+                "STATUS_NO_UPDATE" => xlt("No Change"),
+                "STATUS_OTHER" => xlt("Other"),
+                'APPOINTMENT_STATUS_UPDATE_FAILED' => xlt('There was an error in saving the telehealth appointment status.  Please contact support or update the appointment manually in the calendar')
```

Comment:
not sure where these will all end up, but generally do the escaping when it is echoed/printed since never know if it will need text() or attr()

Reply:
These are used in the frontend on the javascript side so I wanted to escape it here as there's not a good place to escape it on the javascript side as far as I saw.

Reply:
sounds good. if you get in a quagmire with dealing with xlt versus xla (ie. outside and inside html tags) versus context (html versus js where would just need json_encode/js_escape), then there are several javsascript functions openemr has (jsText() and jsAttr())

Reply:
if escape html that goes to js context then some characters can get mangled (also doing html escaping more than once can break things), which is why generally try to do the escaping as far downstream as possible. In this case, would do the following:
1. Change all above from xlt to xl in the above $translations array
2. Change line 3 of interface/modules/custom_modules/oe-module-comlink-telehealth/templates/comlink/telehealth-frontend-settings.js.twig script to (json_encode makes it safe to echo in js context and safe to use as literals in js context):
```
comlink.settings = {{ settings|json_encode }};
```
3. Run the variable in line 895 of interface/modules/custom_modules/oe-module-comlink-telehealth/public/assets/js/telehealth.js through the jsText() function. I am not sure how that line will look (not sure what the backtics are doing), but maybe something like (jsText basically does the same as php text() call and makes it safe to echo into html via the js innerHTML call):
```
<h5 class="modal-title">` + jsText(translations.TELEHEALTH_MODAL_TITLE) + `</h5>
```

Reply:
I've implemented this change in my latest push.  Tested it with a telehealth session and everything is working with the changes from xlt => xl and with the other recommended changes.